src/fish_tests.cpp

```diff
@@ -1033,10 +1034,10 @@ static void test_parser() {
 static void test_1_cancellation(const wchar_t *src) {
     auto filler = io_bufferfill_t::create(io_chain_t{});
     pthread_t thread = pthread_self();
-    double delay = 0.25 /* seconds */;
+    auto delay = static_cast<long int>(0.25 * 1E6) /* seconds */;
```

Comment:
I'm not too sure of this cast. I tried using std:chrono::duration and std::chrono::microseconds to no avail.

With GCC, it reduces to long int, but maybe it reduces to long long int somewhere.

Reply:
Actually yeah. On 32-bit systems.